engine/gamesys/src/gamesys/gamesys_private.cpp

```diff
@@ -83,8 +83,7 @@ namespace dmGameSystem
                 value = &value[value_index];
             }
 
-            out_desc.m_ArraySize = constant_array_size;
-            out_desc.m_IsArray   = constant_array_size > 1;
+            out_desc.m_ValueType   = (constant_array_size > 1) ? dmGameObject::PROP_VALUE_ARRAY : dmGameObject::PROP_VALUE_VARIABLE;
```

Comment:
Now when I look at it, I wonder: Do we really need/want the "PROP_VALUE_VARIABLE"?
A variable can just as easily be represented by an array of 1 element.
The only special case I can think of is get/set of a variable with no "index" specified.

But, we can cover that case using 0 as default index if there is no "index" specified.


That would reduce the 3 types down to 2, which arguably is a lot easier to wrap your head around.

What do you think?

Reply:
done